plugins/destination/plugin_testing.go

```diff
@@ -62,6 +66,29 @@ type PluginTestSuiteTests struct {
 	MigrateStrategyAppend    MigrateStrategy
 }
 
+func RecordDiff(l arrow.Record, r arrow.Record) string {
+	var sb strings.Builder
+	if l.NumCols() != r.NumCols() {
+		return fmt.Sprintf("different number of columns: %d vs %d", l.NumCols(), r.NumCols())
+	}
+	if l.NumRows() != r.NumRows() {
+		return fmt.Sprintf("different number of rows: %d vs %d", l.NumRows(), r.NumRows())
+	}
+	for i := 0; i < int(l.NumCols()); i++ {
+		s, err := array.DiffString(l.Column(i), r.Column(i), memory.DefaultAllocator)
```

Comment:
Just FYI, based on the review I think this function signature will change. But probably not by much.